ams/api/src/main/java/com/netease/arctic/ams/api/process/PendingInput.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.ams.api.process;
+
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+
+import java.util.Set;
+
+public class PendingInput {
```

Comment:
Should we consider to make `PendingInput` immutable?

Reply:
I don't get the benefits and I'm not sure if there could be a wrapper class in the future.

Reply:
Once a pendingInput is created, its state should not be updated. This has the advantage of being thread-safe and easy to maintain. If the pendingInput was mutable, I think it would be complicated to maintain the correct state. WDYT?

Reply:
That make sense
I shall make all fields final

Reply:
Unfortunately

There need be a non-paramenter constructor for recovering from persistence, so final is unmarkable on each field.
But I remove every single set methods. This would have the same effect.

Reply:
Make sense. Also we can add a builder class or annotation to provide flexible constructor